binder/mapping.go

```
@@ -261,10 +262,14 @@
 	switch v := any(value).(type) {
 	case string:
-		assignBindData(out, data, key, v, enableSplitting)
+		assignBindData(out, any(data).(map[string][]string), key, v, enableSplitting)
 	case []string:
 		for _, val := range v {
-			assignBindData(out, data, key, val, enableSplitting)
+			assignBindData(out, any(data).(map[string][]string), key, val, enableSplitting)
+		}
+	case []*multipart.FileHeader:
+		for _, val := range v {
+			data[key] = append(data[key], any(val).(T))
```

golangci-lint (1.62.2)

type assertion must be checked

(forcetypeassert)